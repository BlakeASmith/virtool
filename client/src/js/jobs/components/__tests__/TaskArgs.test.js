import {
    AnalysisRows,
    BuildIndexRows,
    CreateSampleRows,
    CreateSubtractionRows,
    TaskArgsRows,
    UpdateSampleRows
} from "../TaskArgs";

const taskTypes = ["aodp", "build_index", "create_sample", "create_subtraction", "pathoscope_bowtie", "nuvs", "update_sample"];

describe("<TaskArgs />", () => {
    let props;

    it.each(taskTypes)("renders <TaskArgsRows /> correctly when task type is %p", taskType => {
        props = {
            taskType,
            taskArgs: {
                sample_id: "123abc",
                analysis_id: "test-analysis"
            }
        };
        const wrapper = shallow(<TaskArgsRows {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<AnalysisRows />", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<AnalysisRows analysis_id="foo" sample_id="bar" />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<BuildIndexRows />", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<BuildIndexRows ref_id="foo" index_id="bar" />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<CreateSampleRows />", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<CreateSampleRows sample_id="foo" />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<CreateSubtractionRows />", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<CreateSubtractionRows subtraction_id="foo" />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<UpdateSampleRows />", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<UpdateSampleRows sample_id="foo" />);
        expect(wrapper).toMatchSnapshot();
    });
});
