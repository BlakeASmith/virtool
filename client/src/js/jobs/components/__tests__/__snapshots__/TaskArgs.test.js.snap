// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalysisRows /> renders correctly 1`] = `
<Fragment>
  <TaskArgsRow
    title="Sample"
  >
    <Link
      to="/samples/bar"
    >
      bar
    </Link>
  </TaskArgsRow>
  <TaskArgsRow
    title="Analysis"
  >
    <Link
      to="/samples/bar/analyses/foo"
    >
      foo
    </Link>
  </TaskArgsRow>
</Fragment>
`;

exports[`<BuildIndexRows /> renders correctly 1`] = `
<Fragment>
  <TaskArgsRow
    title="Reference"
  >
    <Link
      to="/refs/foo"
    >
      foo
    </Link>
  </TaskArgsRow>
  <TaskArgsRow
    title="Index"
  >
    <Link
      to="/refs/foo/indexes/bar"
    >
      bar
    </Link>
  </TaskArgsRow>
</Fragment>
`;

exports[`<CreateSampleRows /> renders correctly 1`] = `
<Fragment>
  <TaskArgsRow
    title="Sample"
  >
    <Link
      to="/samples/foo"
    >
      foo
    </Link>
  </TaskArgsRow>
</Fragment>
`;

exports[`<CreateSubtractionRows /> renders correctly 1`] = `
<TaskArgsRow
  title="Subtraction"
>
  <Link
    to="/subtraction/foo"
  >
    foo
  </Link>
</TaskArgsRow>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "aodp" 1`] = `
<AnalysisRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "build_index" 1`] = `
<BuildIndexRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "create_sample" 1`] = `
<CreateSampleRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "create_subtraction" 1`] = `
<CreateSubtractionRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "nuvs" 1`] = `
<AnalysisRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "pathoscope_bowtie" 1`] = `
<AnalysisRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<TaskArgs /> renders <TaskArgsRows /> correctly when task type is "update_sample" 1`] = `
<UpdateSampleRows
  analysis_id="test-analysis"
  sample_id="123abc"
/>
`;

exports[`<UpdateSampleRows /> renders correctly 1`] = `
<TaskArgsRow
  title="Subtraction"
>
  <Link
    to="/samples/foo"
  >
    foo
  </Link>
</TaskArgsRow>
`;
